crates/pallet-domains/src/staking.rs

```diff
@@ -38,33 +38,48 @@ pub(crate) struct Deposit<Share: Copy, Balance: Copy> {
 /// A share price is parts per billion of shares/ai3.
 /// Note: Shares must always be equal to or lower than ai3.
 #[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq, Default)]
-pub struct SharePrice(Perbill);
+pub struct SharePrice(Option<Perbill>);
 
 impl SharePrice {
     /// Creates a new instance of share price from shares and stake.
-    pub(crate) fn new<T: Config>(shares: T::Share, stake: BalanceOf<T>) -> Self {
-        SharePrice(if shares.is_zero() || stake.is_zero() {
-            Perbill::one()
+    /// Returns an error if there are more shares than stake. If either value is zero, the share
+    /// price is marked as missing internally.
+    pub(crate) fn new<T: Config>(shares: T::Share, stake: BalanceOf<T>) -> Result<Self, Error> {
+        if shares > stake.into() {
```

Comment:
Wondering if there is possibility of this happeneing.
Though shares should always be be less than stake unlike bundle storage fund